### ISSUE TITLE
Sound disappears while placing a block

### ISSUE THREAD

**Issue by vlouboos** [Foreheadchann/ViaMCP-Reborn#22]
I set my protocol version to 340(1.12.2),and entered an 1.12.2-Server.
If I use a 1.12.2 client, the sound plays normally.
While I was using a 1.8.9-based-client to place blocks, the sound disappears.
However, when I use this client to enter a 1.8.9-Server, the sound plays normally again.
Is this a bug that can be fixed or a problem?

**Comment by vlouboos** [comment]
Not like the issue before on January, the map's been fixed.
The sound could be played while attacking others on the 1.12.2-server, but not while placing blocks.

**Comment by Foreheadchann** [comment]
Blockplacement sound mappings are always broken and need to be fixed client side, as the server sends a different sound path for placed blocks (some other sounds may be broken too, but block placing is the most known)

**Comment by vlouboos** [comment]
Actually, it's not the mappings' fault. I debugged the sound system. and the result is: NO SOUND DATA WAS BACK FROM SERVER WHILE PLACING A BLOCK BY MYSELF. 
By the way, I found that I do hear the sound while other players place blocks.
![image](https://user-images.githubusercontent.com/87124020/153550341-6f8c4850-b2de-4f88-ab53-eb9be3e92cea.png)
![image](https://user-images.githubusercontent.com/87124020/153550402-48fc679a-07de-4577-9605-45494dc63b9d.png)

**Comment by vlouboos** [comment]
Ummm...
The "client"s above mean "Protocol"

**Comment by Foreheadchann** [comment]
That's weird, as usually (before I removed the debugging for it) it would display Unknown Sound on 1.8+ servers using 1.8 client. I will try to look into this later and try to find a fix, if even possible on client side. It could be, that they changed the packet format for play sound (Which should be S29 if I'm not mistaken), so the client receives an invalid packet, but I'm not really sure of that, as it should throw a NullPointerException when received (Like some servers when they send for example S3E packets).

TLDR: Will look into it, help appreciated :)

**Comment by vlouboos** [comment]
I'm also confused.However, unfortunately, it didn't throw any exception.

**Comment by Foreheadchann** [comment]
Progress Report, and possible cause

Tested function destroyBlock(BlockPos,boolean) in World, and added playing fx in if clause where check if block material == air, which would normally return false only, but for some reason, when adding sounds also in the is air check, all block breaking sounds are fixed, still need to find placement sounds

**Comment by Foreheadchann** [comment]
I am not at all sure why all block materials would be returned as Material.air

**Comment by vlouboos** [comment]
Maybe it's the order of behavior & packet sending.
I found something other changed before, like in 1.8, players swing hands first and then attack. But in higher version, players attack first, then swing.
I wonder if this is the same. Higher-version-server requires blocks reporting before client gives placing action, but client may first give placing action, then report the block.

**Comment by Foreheadchann** [comment]
I would also appreciate if you know where the block placement function is, as I cannot find it with the sound effect id (2001) or playAusSFX

**Comment by Foreheadchann** [comment]
I have found it, and you are very much correct, it is a packetorder issue, I will be releasing a fix for the 1.8.x base asap!

**Comment by Foreheadchann** [comment]
Seems like it was not the way i found, as somehow my version was reset to 1.8.x for no reason, and as such was playing placement sounds, still need to look more

**Comment by Foreheadchann** [comment]
Block Breaking and Placing sound effects have now an easy fix method in the README, thank you for letting me know, that the mappings were not the only issue
